code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp

```diff
@@ -0,0 +1,283 @@
+            *leftScanRegion_, *rightScanRegion_,
+            kvConfig.getValue<ColumnIndex>(MODULE_NAME + ".maxlanemarkingwidth"),
+            kvConfig.getValue<ColumnIndex>(MODULE_NAME + ".maxmarkingrowcount"),
+            kvConfig.getValue<double>(MODULE_NAME + ".maxmarkinginclination"),
+            kvConfig.getValue<size_t>(MODULE_NAME + ".minpixelcount")));
+
+        if(DEBUG_IS_SET) {
+            cv::namedWindow(MODULE_NAME);
+            cv::setMouseCallback(MODULE_NAME, onMouse);
+        }
+    }
+
+    void nextContainer(Container & container) {
+        if(SharedImage::ID() == container.getDataType()) {
+            SharedImage image = container.getData<SharedImage>();
+
+            if(isDesiredSharedImage(image)) {
+                if(!attachToSharedMemory(image)) {
+                    return;
+                }
+
+                cv::Mat cvImage = sharedImageAdapter_->fetchImage();
+                processImage(cvImage);
+            }
+        }
+    }
+
+    void tearDown() {
+        if(DEBUG_IS_SET) {
+            cv::destroyWindow(MODULE_NAME);
+        }
+    }
+
+private:
+    static const std::string MODULE_NAME;
+    static const std::string CONFIG_NAME;
+
+    /**
+     * Run upon construction to fetch the desired sharedImage name from the configuration
+     *
+     * @return The sharedImageName configuration value. Empty string if not set.
+     */
+    std::string getSharedImageName() {
+        try {
+            return getKeyValueConfiguration().getValue<std::string>(MODULE_NAME + ".sharedimagename");
+        }
+        catch(const ValueForKeyNotFoundException & e) {
+            CLOG << "Didn't find key " << MODULE_NAME + ".sharedimagename" << std::endl;
+            return std::string();
+        }
+    }
+
+    /**
+     * Checks whether the given image has the name set in sharedImageName_. If it is not set, it is
+     * overwritten with the name of the given shared image, so that this is used in the future.
+     */
+    bool isDesiredSharedImage(const SharedImage & image) {
+        if(sharedImageName_.empty()) {
+            CLOG << "no \"sharedImageName\" config value set, using shared image " << image.getName() << std::endl;
+            sharedImageName_ = image.getName();
+            return true;
+        }
+
+        return 0 == image.getName().compare(sharedImageName_);
+    }
+
+    /**
+     * Try to attach to the shared memory, initializing the sharedImageAdapter_ member.
+     *
+     * @return true if attaching succeeded or already done, false if it failed
+     */
+    bool attachToSharedMemory(const SharedImage & image) {
+        if(!sharedImageAdapter_) {
+            try {
+                sharedImageAdapter_.reset(new SharedImageCvAdapter(image));
+            }
+            catch(const FailedAttachingToSharedMemoryException & e) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+#ifndef NDEBUG
```

Comment:
Please let this function be part of the regular source tree and use if based on the value of your DEBUG_IS_SET value.